src/audio/module_adapter/module_adapter.c

```diff
@@ -23,6 +23,19 @@
 
 LOG_MODULE_REGISTER(module_adapter, CONFIG_SOF_LOG_LEVEL);
 
+/*
+ * helpers to determine processing type
+ * Needed till all the modules use PROCESSING_MODE_SINK_SOURCE
+ */
+#define IS_PROCESSING_MODE_AUDIO_STREAM(mod) \
+		(!!((struct module_data *)&(mod)->priv)->ops->process_audio_stream)
+
+#define IS_PROCESSING_MODE_RAW_DATA(mod) \
+		(!!((struct module_data *)&(mod)->priv)->ops->process_raw_data)
+
+#define IS_PROCESSING_MODE_SINK_SOURCE(mod) \
+		(!!((struct module_data *)&(mod)->priv)->ops->process)
```

Comment:
sorry, I'm a bit confused. An earlier patch in this PR added sink and source APIs to the "audio stream API" (i.e. `struct audio_stream`). That means that all module adapter users, setting the `simple_copy` flag can now already directly be switched to using the sink / source API. So you shouldn't need separate `module_adapter_audio_stream_type_copy()` and `IS_PROCESSING_MODE_AUDIO_STREAM()`?